### DISCUSSION TITLE
Generating User/Developer personas

### DISCUSSION THREAD

**Discussion by annakrystalli** [hubverse-org/hubUtils#11] [Ideas]
Given I'm not intimately familiar (yet) with the various use cases and stakeholders and software user and developer profiles, it feel it would be very useful to go through a formal process of creating user personas: https://the-turing-way.netlify.app/project-design/persona/persona-creation.html

Some examples: https://socialimpact.github.com/assets/img/research/GitHub_OSSSocialSector_ProtoPersonas_v2.pdf

This could be something that we can start to put together but then discuss and review during one of the consortium calls. What do you think @nickreich & @elray1

**Comment by annakrystalli** [comment]
In effect the exercise aims to define more explicitly the principles, requirements and scope of Hub software by translating the general needs specified in for example the following Hub principle statement  into needs of specific personas with specific roles and attributes. 
> -  Usability: Modeling hubs strive to build infrastructure suitable for diverse stakeholders engaging in different stages of the hub processes. Hub coordinators, teams submitting projections, consumers of models, researchers interested in conducting secondary analyses, and other potential users should have access to tools suitable to their expected level of expertise. Although modelers and researchers might safely be assumed to be familiar with modern software development tools and popular platforms, consumers of models might be more comfortable with spreadsheets and GUI interfaces. Modeling hubs should strive to develop tools that will be usable by the intended audience, avoiding barriers that might prevent or discourage participation or use of data. 


Personas generated should cover the full spectrum of stakeholders, their level of engagement with the software and needs.

**Comment by nickreich** [comment]
It's not clear to me how varied the use cases are here, given the fairly basic set of functionality that we are hoping to focus on. I see most users as falling into a pretty clear bucket of a "quantitative analyst" of some kind. My quick sketch is:
 - someone already familiar with R, maybe a power-user, but we shouldn't assume advanced familiarity with object-oriented programming
 - wants to run some analyses with forecast data, like make some plots, score some forecasts, etc...

**Comment by annakrystalli** [comment]
Thanks for the input @nickreich ! Sorry I probably didn't explain the exercise well enough to explain the detail, value and goals. 

Just in the the Usability section of the hub software principles I mentioned above, there are a number of different roles / stakeholders that may have different needs & potential experience levels (which could be expressed as barriers). For example:
- Hub coordinators, 
- teams submitting projections, 
- consumers of models, 
- researchers interested in conducting secondary analyses.

Scenario modelling vs forecasting modelling is also already also exposing a number of differing user needs.

I can imagine funders who might want to report impact of software or policy makers (perhaps that's what you mean by consumers of models?) having certain needs also.

In the scope document you also list the aims of the software to be to support **creating, hosting, maintaining, and running a modeling hub** as well as the potential for the the tools developed as part of this effort to be designed to be more general and be able to be used for other purposes, e.g., for aggregating estimates of parameters of interest.

It is best to assume that for example people creating a hub may not be the ones that end up maintaining them so this, for instance, might affect the type of documentation a new maintainer would need to continue supporting and updating a given. Also how are we proactively envisaging and taking into account hubs being used for other purposes?

Each of these separate user/engagement scenarios would (ideally!) get a separate persona detailing their particular attributes, goals, needs and potential barriers to use.

Personas can be extremely detailed but I suggest we keep it simple by focusing on just a few attributes:
- A name
- A persona description which would include for example, affiliation role and general use profile
- Goals: List what goals the persona needs to achieve with the software.
- Needs: which would list the particular wants & needs the user requires from the software to achieve their goals
- an optional barriers section which could include things like unfamiliarity with tools, language barriers etc.

I'll provide a template below that consortium members can use to submit personas should they wish to participate. I feel that if everyone provided one or two personas (and try and think of some of the stakeholders that are more distantly related to the consortium group), we could compile some good user profiles.

I know it can seem a bit awkward/superfluous but I've been forced to do enough of these already that I'm now convinced of their value. It doesn't mean that we can necessarily meet all needs of all possible stakeholders but it's always good to be aware of them. In any case, we reduce the risk of development going in a direction that makes it harder to accommodate or completely ignores the needs of stakeholders that the developer can't necessarily envisage. Given my own relative unfamiliarity still of different stakeholders and use cases the risk of this is higher currently. it also means I'd understand what is being discussed in the calls a lot better and be able make more informed decisions when developing rather than have to constantly have to ask questions (although input will always be sought at the pull request review stage).

As always, thoughts and feedback welcome!

```
# Modeling Hub Software Stakeholder Persona
## Name:
*Give the persona a name*

## Persona Description

*Describe persona including affiliation, professional role, experience with relevant tools and any other characteristic that might be important to the user profile.*

## Goals
*List what goals the persona needs to achieve with the software.*

## Needs
*List list the particular wants & needs the user requires from the software to achieve their goals.*

## Barriers
*List any barriers you could envisage that could creating problems for the user in meeting their goals.*

```
